### PR TITLE
feat(Zone): check global for rootZoneSpec

### LINKED ISSUE #513: Configuring Root zone
Is there a way to configure `onScheduleTask` and `onInvokeTask` on the root zone?
Or other interceptions for that matter?

### PR DESCRIPTION
**Pull request by jahtalab**
closes #513

### CODE REVIEW

**Comment by JiaLiPassion** [comment]
As described in comment of #513,  I am not sure my thinking is correct or not, but I think we shouldn't allow access ZoneSpec at any level.

File: lib/zone.ts [hunk 1]
```diff
@@ -952,7 +952,14 @@ const Zone: ZoneType = (function(global: any) {
   const symbolPromise = __symbol__('Promise');
   const symbolThen = __symbol__('then');
 
-  let _currentZoneFrame = new ZoneFrame(null, new Zone(null, null));
+  const symbolRootZoneSpec = '__rootZoneSpec__'
+  var rootZone: Zone = new Zone(null, null);
```
Comment (JiaLiPassion) [inline lib/zone.ts]: We should use let instead of var

File: lib/zone.ts [hunk 2]
```diff
@@ -952,7 +952,13 @@ const Zone: ZoneType = (function(global: any) {
...
+  if (global[symbolRootZoneSpec]) {
+    rootZone = <Zone>rootZone.fork(global[symbolRootZoneSpec]);
```
Comment (jahtalab) [inline lib/zone.ts]: I think we should so that we don't clutter global space.
I can't see any reason not to.

**Comment by jahtalab** [comment]
Could you please rerun the tests on travis?
I'm not sure what is causing the failure.

**Comment by JiaLiPassion** [comment]
Sorry, I have no permission to rerun the test on travis, the error seems to be timeout, it should be ok to run again, in your PR, a ';' is missing in line 955, maybe you can commit the ';' and it will trigger a new travis test run.

**Comment by JiaLiPassion** [comment]
It seems to be the fetch issue, I got the same problem before, I'll check it later.

**Comment by jahtalab** [comment]
just rebased, tests are passing now. can we merge this one?

**Comment by mhevery** [comment]
Sorry, but this is by design. RootZone should be indistinguishable for no zone at all. This requirement is here so that if we standardize the Zones in TC39, the browser vendors will not have to pay the cost for all of the applications. All zones must be entered explicitly.

**Pull request closed without merging**